network/evpn-gw/evpn-gw-cni/pkg/types/types.go

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright (C) 2023 Network Plumping Working Group
+// Copyright (C) 2023 Nordix Foundation.
+
+package types
+
+import (
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/vishvananda/netlink"
+)
+
+// VfState represents the state of the VF
+type VfState struct {
```

Comment:
What's the utility of SRIOV cni? Why cannot we use the opensourced one?
https://github.com/openshift/sriov-cni/blob/master/pkg/sriov/sriov.go#L220

Reply:
Those features currently are not supported by the IPU. Maybe will in the future. Also is not common for one CNI to import from another one. The CNIs entities are mostly standalone and are tailormade mostly for a specific networking object. In this case SRIOV cni is more tailormade for Legacy SRIOV cards and not for Smartnics. The EVPN GW CNI supports IPU for EVPN GW usecase. In the future we will have an OPI-CNI that will support all the usecases on a IPU/DPU offload as well as EVPN GW usecase.